### PR TITLE
Feature/ted 84

### PR DESCRIPTION
**Pull request by CaptainOfHacks**

### CODE REVIEW

File: tests/unit/data_manager/conftest.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import mongomock
+import pymongo
+import pytest
+
+
+@pytest.fixture
+@mongomock.patch(servers=(('server.example.com', 27017),))
```
Comment (costezki) [inline tests/unit/data_manager/conftest.py:7]: We are gonna have to decide whether we use `Fakes` or `mock`s.
Generally, we do not need both unless I am ignorant or missing something.

File: tests/unit/data_manager/test_notice_repository.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from ted_sws.data_manager.adapters.notice_repository import NoticeRepository
+from ted_sws.domain.model.manifestation import XMLManifestation
+from ted_sws.domain.model.metadata import TEDMetadata
+from ted_sws.domain.model.notice import Notice
+
+NOTICE_TED_ID = "123456"
+TEST_DB_NAME = 'test_db'
+
+
+def test_notice_repository_create(mongodb_client):
```
Comment (costezki) [inline tests/unit/data_manager/test_notice_repository.py:12]: This test is identical to the other one in e2e.

**Review by costezki — CHANGES_REQUESTED** [review]

File: ted_sws/data_manager/adapters/notice_repository.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        notice_dict["_id"] = notice_dict["ted_id"]
+        self.collection.insert_one(notice_dict)
+
+    def get(self, reference) -> Notice:
+        """
+            This method allows a notice to be obtained based on an identification reference.
+        :param reference:
+        :return: Notice
+        """
+        result_dict =  self.collection.find_one({"ted_id": reference})
```
[the remark this answers is not in the document]
Comment (CaptainOfHacks) [reply ted_sws/data_manager/adapters/notice_repository.py:37]: _id == ted_id

File: ted_sws/data_manager/adapters/notice_repository.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        result_dict = self.collection.find_one({"ted_id": reference})
+        return Notice(**result_dict) if result_dict else None
+
+    def list(self) -> Iterator[Notice]:
+        """
+            This method allows all records to be retrieved from the repository.
+        :return: list of notices
+        """
+        for result_dict in self.collection.find():
+            yield Notice(**result_dict)
```
Comment (costezki) [inline ted_sws/data_manager/adapters/notice_repository.py:55]: Clever! 

I only hope that `self.collection.find()` is also a generator.

**Review by costezki — APPROVED** [review]
Looking good!

**Comment by costezki** [comment]
BTW the tests are failing :)

File: ted_sws/data_manager/adapters/notice_repository.py (hunk 4 above)
Comment (CaptainOfHacks) [reply ted_sws/data_manager/adapters/notice_repository.py:55]: yes

**Pull request merged**